src/prefect/client/secrets.py

```diff
@@ -74,6 +74,23 @@ class Secret:
     def __init__(self, name: str):
         self.name = name
 
+    def exists(self) -> bool:
+        """
+        Determine if the secret exists.
+
+        Returns:
+            - bool: a boolean specifying whether the Secret is accessible or not
+        """
+        secrets = prefect.context.get("secrets", {})
+        if self.name in secrets:
+            return True
+        elif prefect.context.config.cloud.use_local_secrets is False:
+            client = Client()
```

Comment:
I'm wondering if it would be more efficient to create the `Client` object when the class __init__ is called, rather than creating it on each method invocation?

Reply:
Can do - will update

Reply:
Updated